src/test/scala/com/microsoft/hyperspace/index/HybridScanSuite.scala

```diff
@@ -396,32 +396,37 @@ trait HybridScanSuite extends QueryTest with HyperspaceSuite {
         val query = df2.filter(df2("clicks") <= 4000).select(df2("clicks"), df2("Date"))
         query2.join(query, "clicks")
       }
-      val baseQuery = joinQuery()
-      val basePlan = baseQuery.queryExecution.optimizedPlan
 
-      withSQLConf("spark.sql.autoBroadcastJoinThreshold" -> "-1") {
-        withSQLConf(
-          SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> "false",
-          IndexConstants.INDEX_HYBRID_SCAN_ENABLED -> "false") {
-          val join = joinQuery()
-          checkAnswer(join, baseQuery)
-          assert(basePlan.equals(join.queryExecution.optimizedPlan))
-        }
+      // Dynamic pruning creates a dynamic filter, with different ids every time
+      // thus making basePlan.equals(join) fail
+      withSQLConf("spark.sql.optimizer.dynamicPartitionPruning.enabled" -> "false") {
```

Comment:
👍 

You can change the config in `beforeAll` or `before` in HybridScanForPartitionedDataTest

Reply:
yep seems so as long as there's no exception in after() https://github.com/yyuu/scalatest/blob/master/src/main/scala/org/scalatest/BeforeAndAfter.scala